clippy_lints/src/assign_ops.rs

```diff
@@ -135,6 +135,19 @@ impl LateLintPass for AssignOps {
                                         } else {
                                             return; // useless if the trait doesn't exist
                                         };
+                                        // check that we are not inside an `impl AssignOp` of this exact operation
+                                        let parent_fn = cx.tcx.map.get_parent(e.id);
+                                        let parent_impl = cx.tcx.map.get_parent(parent_fn);
+                                        if parent_impl.as_u32() != 0 {
```

Comment:
That's kind of a feature, `Option<NodeId>` is not necessary as there is `syntax::ast::DUMMY_NODE_ID`. Yup, that's weak typing, but it's more memory efficient, and `NodeId`s are used everywhere.